### DISCUSSION TITLE
Difference b/w recordings

### DISCUSSION THREAD

**Discussion by vajrang-b** [vajrang-b/RPA-Developer-in-30-Days#16] [Q&A] [answered]
Basic
Desktop
Web 
Image

**Comment by usurabhi** [comment, marked as answer]
**Basic** : Each activity in the basic recorder generates full selectors, In this recording it cannot generate container for every activity, it is slower than the basic desktop recording, it is suitable for single activity
**Desktop**: It generates a partial selectors, it generates a container for each activity, it is faster than the basic recorder, it is suitable for more than one activity
**Web** : Designed for recording in web apps and browsers, generates containers and uses the simulate type/click input methods by default
**Image** : It permits only image, text and keyboards automations and requires explicit positioning
Citrix: Is the equivalent of the Desktop recorder, but for Citrix environments. Only use this in your Native Citrix automation projects.

**Comment by TiwariRakesh** [comment]
Basic recorder- it uses full selector(Root node and Parent node), it slow also suitable for linear process automation.
Desktop recorder- it uses partial selector(Doesn't care about root node), faster than Basic recorder, suitable for desktop and web automation.
Web- suitable for automating web application.
image- suitable for automating image and text.

**Comment by Ulhasguttikar** [comment]
**Basic** :- It will Create full Selectors. Slower than recorders which use Partial Selectors.
**Desktop** :-  Used for Desktop Applications. Generate Partial Selector. Faster than Basic Recorder.
**Web** :- Used for Web recording. Generates Partial Selector. uses the Simulate Type/Click input method by default.
**Image* :- Used to record virtualized environments (such as VNC, virtual machines, Citrix, and more) or SAP. It permits only image, text and keyboard automation, and requires explicit positioning.
